<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>529</width>
    <height>336</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>529</width>
    <height>336</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Console</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget {
	background-color: white;
	font-family: cantarell;
	font-size: 13px;
}

QLabel {
	font-size: 16px;
}

QPushButton {
    border: 1px solid black; /* Black border */
    background-color: lightgray; /* Light gray background */
	font-weight: bold;
}

QPushButton:hover {
    background-color: darkgray; /* Slightly darker background on hover */
}

QPushButton:pressed {
    background-color: gray; /* Even darker background when pressed */
}

QGroupBox {
	font-size: 20px;
    border: none; /* Removes the default border */
    border-top: 1px dashed black; /* Adds a border under the title */
    margin-top: 30px; /* Ensures space for the title */
    padding-top: 10px; /* Creates space above the border for the title */
}

QGroupBox::title {
    subcontrol-origin: margin; /* Aligns the title with the margin */
    subcontrol-position: top left; /* Moves the title to the top-left corner */
    padding: -5px; /* Adds padding around the title */
    background-color: white; /* Optional: matches the background */
}

QTableView {
	border: none;
}

QComboBox {
    background-color: white; /* Default background */
    color: black; /* Default text color */
    border: 1px solid black; /* Border around the combobox */
    padding: 2px 10px; /* Padding for text alignment */
}

QComboBox QAbstractItemView {
    background-color: lightgray; /* Background of the dropdown */
    color: black; /* Text color of items in the dropdown */
    selection-background-color: darkgray; /* Background color when an item is selected */
    selection-color: white; /* Text color when an item is selected */
}
</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_4">
   <item row="0" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="0">
     <item>
      <widget class="QPlainTextEdit" name="plainTextEditSerialOutput">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>100</height>
        </size>
       </property>
       <property name="lineWrapMode">
        <enum>QPlainTextEdit::NoWrap</enum>
       </property>
       <property name="readOnly">
        <bool>false</bool>
       </property>
       <property name="placeholderText">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <layout class="QGridLayout" name="gridLayout">
     <item row="1" column="1">
      <widget class="QSpinBox" name="spinBoxLimitLines">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximum">
        <number>10000</number>
       </property>
       <property name="value">
        <number>5000</number>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLabel" name="label_6">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Lines</string>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QCheckBox" name="checkBoxDebugCursorFollow">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Cursor follow</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="4">
      <widget class="QPushButton" name="pushButtonClear">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>94</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>Clear</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0" rowspan="2">
      <widget class="QCheckBox" name="checkBoxLimitLines">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Keep maximum of</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="2" column="0">
    <widget class="QGroupBox" name="groupBox_16">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>140</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>140</height>
      </size>
     </property>
     <property name="title">
      <string>Filter messages</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <layout class="QGridLayout" name="gridLayout_3">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="QCheckBox" name="checkBoxFilterCommAck">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Command aknowledge</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QCheckBox" name="checkBoxFilterUndefined">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Undefined messages</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QCheckBox" name="checkBoxFilterInfoRequest">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Info Requests</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QCheckBox" name="checkBoxFilterUSB">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>USB Commands</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QCheckBox" name="checkBoxFilterPriority">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Priority messages</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QCheckBox" name="checkBoxFilterExpressionParser">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Expression parser</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0" rowspan="2">
         <widget class="QCheckBox" name="checkBoxFilterHardware">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Hardware messages</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QCheckBox" name="checkBoxFilterError">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Error messages</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2" rowspan="2">
         <widget class="QCheckBox" name="checkBoxFilterDebug">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Debug messages</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QCheckBox" name="checkBoxFilterOutput">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Output messages</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QLineEdit" name="lineEditSend"/>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="pushButtonSend">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>94</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>Send</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
